client/engine/store/memstore.go

```diff
@@ -179,6 +179,37 @@ func (ms *MemStore) getChannelById(id types.Destination) (channel.Channel, error
 	return ch, nil
 }
 
+// GetChannelsByIds returns a collection of channels with the given ids
+func (ms *MemStore) GetChannelsByIds(ids []types.Destination) ([]*channel.Channel, error) {
+	toReturn := []*channel.Channel{}
+
+	var err error
+
+	ms.channels.Range(func(key string, chJSON []byte) bool {
+		var ch channel.Channel
+		err = json.Unmarshal(chJSON, &ch)
+		if err != nil {
+			return false
+		}
+
+		// If the channel is one of the ones we're looking for, add it to the list
+		if contains(ids, ch.Id) {
```

Comment:
Nit: I suppose there is an opportunity for optimization, here: no need to check the `ch.Id` against _all_ passed `ids` since any we have already found may be disregarded.

Reply:
Good point. Whenever we find a channel we could remove it from `ids`